src/sonic-yang-models/yang-models/sonic-nat.yang

```diff
@@ -0,0 +1,281 @@
+module sonic-nat {
+    namespace "http://github.com/Azure/sonic-nat";
+    prefix snat;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-acl {
+        prefix sacl;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC NAT";
+
+    revision 2019-10-30 {
+        description
+            "Initial revision.";
+    }
+
+    typedef nat-clear-param {
+        type enumeration {
+            enum ENTRIES {
+                description "NAT dynamic Entries";
+            }
+            enum STATISTICS {
+                description "NAT Statistics";
+            }
+        }
+    }
+
+    typedef nat-type {
+        type enumeration {
+            enum snat {
+                description "Source NAT.";
+            }
+            enum dnat {
+                description "Destination NAT.";
+            }
+        }
+    }
+
+    typedef ip-protocol-type {
+        type enumeration {
+            enum TCP;
+            enum UDP;
+        }
+    }
+
+    grouping sonic-twice-nat-id {
+
+        leaf twice_nat_id {
+            type uint16 {
+                range "1..9999" {
+                    error-message "Invalid Group Id for twice NAPT.";
+                    error-app-tag twice-nat-id-invalid;
+                }
+            }
+        }
+    }
+
+    grouping sonic-nat-common {
+
+        leaf global_ip {
+            type inet:ipv4-address;
+        }
+
+        leaf local_ip {
+            mandatory true;
+            type inet:ipv4-address;
+        }
+
+        leaf nat_type {
+            type nat-type;
+            default dnat;
+        }
+        uses sonic-twice-nat-id;
+    }
+
+    typedef ip-address {
+        type union {
+          type inet:ipv4-address;
+          type string {
+                          pattern '(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(-(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9]))?';
+                      }
+        }
+
+        description
+          "This type denotes a single IP address or a range of addresses.";
+    }
+
+    container sonic-nat {
+
+        container STATIC_NAPT {
+
+            list STATIC_NAPT_LIST {
+                key "global_ip ip_protocol global_l4_port";
+
+                ext:key-regex-configdb-to-yang "^([a-fA-F0-9:./]+)|(TCP|UDP)|([0-9]+)$";
```

Comment:
I believe, we decided not to define key-regex for the list, @praveen-li We don't need this anymore..correct?

Reply:
Removed key-regex.